### DISCUSSION TITLE
invokeOnewayImpl是否有必要使用SemaphoreReleaseOnlyOnce？

### DISCUSSION THREAD

**Discussion by 2335119327** [apache/rocketmq#6848] [Q&A] [answered]
<img width="996" alt="image" src="https://github.com/apache/rocketmq/assets/61485889/896f3fd7-7c61-4bae-96e5-cee91d63354e">

对于release操作，能否直接使用try-catch-finally？在finally中进行release即可。

**Comment by 2335119327** [comment]
请大佬们指教一下~

**Reply by xiaoyifang** [reply, marked as answer]
> 能否直接使用try-catch-finally？在finally中进行release即可。

应该不能，   看原逻辑，正常release是在异步回调里执行的。    
如果在finally里释放，会造成提前释放。

**Reply by 2335119327** [reply]
soga，感谢指教~
